conf/pulsar_env.sh

```diff
@@ -60,6 +60,10 @@ for token in $("$JAVA_BIN" -version 2>&1 | grep 'version "'); do
           JAVA_MAJOR_VERSION=${BASH_REMATCH[1]}
         fi
         break
+    elif [[ $token =~ \"([[:digit:]]+)(.*)\" ]]; then
```

Comment:
I think it's better to include `-internal` as part of the pattern to make this explicit. 

The version parsing logic is also duplicated in bkenv.sh. That would need to be addressed too.

Reply:
> I think it's better to include -internal as part of the pattern to make this explicit.

I suspect whether there are other special versions, so I use this loose pattern which only identities the first digit number. I don't think it is good to restrict it as `-internal`. But I will adjust the code if @lhotari insists his opinion after second thought.

> The version parsing logic is also duplicated in bkenv.sh. That would need to be addressed too.

I will address `bkenv.sh` soon.

Reply:
The version number syntax is standardized, so why would there be a need for other syntaxes? If a new format is added, we could deal with it if that ever happens.

Reply:
https://openjdk.java.net/jeps/223 .

Reply:
I guess it would make sense if the pattern handles the $PRE format described by the JEP

Reply:
> https://openjdk.java.net/jeps/223 .

Thanks for pointing out the standard. I list the following important information.

```
Version numbers $VNUM
[1-9][0-9]*((\.0)*\.[1-9][0-9]*)*

Version strings $VSTR
$VNUM(-$PRE)?\+$BUILD(-$OPT)?
$VNUM-$PRE(-$OPT)?
$VNUM(+-$OPT)?


Name                            Syntax
------------------------------  --------------
java.version                    $VNUM(\-$PRE)? 
java.runtime.version            $VSTR
java.vm.version                 $VSTR
java.specification.version      $VNUM
java.vm.specification.version   $VNUM

$ java -version
openjdk version \"${java.version}\"
${java.runtime.name} (build ${java.runtime.version})
${java.vm.name} (build ${java.vm.version}, ${java.vm.info})
```

From the standard, we can know the `java.version` is started by a **necessary major version number**, then followed by zero or several "sub-version"(minor/security), and finally ended with a optional `-$PRE`.

So I can confirm that the newly added pattern `\"([[:digit:]]+)(.*)\"` can address all the common cases of the standard. And the pattern `\"([[:digit:]]+)\.([[:digit:]]+)\.(.*)\"` you added previously solves the vein which the `java.version` has at aleast one  "sub-version".

> I guess it would make sense if the pattern handles the $PRE format described by the JEP

If we want to handle the `$PRE` format by changing the pattern `\"([[:digit:]]+)(.*)\"` . I may ask: should we need to change the `\"([[:digit:]]+)\.([[:digit:]]+)\.(.*)\"` to handle the `$PRE` too? If the answer is no, why we need to handle `$PRE` in this new common pattern.

Reply:
From the standard, I find the pattern `\"([[:digit:]]+)\.([[:digit:]]+)\.(.*)\"` can't handle the version like `1.9-XXX`. I remove the unnecessary latest dot `\.`.